spec.md

```diff
@@ -474,6 +474,122 @@ The following string-encoded integer literals all have the value `-42`:
 Radix-specified integer literals are only usable when constructing hardware
 integer literals.  Any use in place of an integer is disallowed.
 
+# Circuit Components
+
+Circuit components are the named objects which may be contained within a module.
+They are: nodes, wires, registers, ports, and submodule instances.
+
+TODO: What about `mem`s?
+
+## Nodes
+
+Nodes are named expressions in FIRRTL.
+They represent a small piece of combinational logic.
+
+Example:
+
+```firrtl
+    node mynode = and(in, UInt<4>(1))
+```
+
+Nodes are given a name and a definition.
+Unlike wires, they can only be assigned to at the point they are defined.
+(TODO: Do all values need to be in scope above the point of definition?)
+They cannot be connected (see [@sec:connects]).
+
+## Wires
+
+A wire is a named combinational circuit component that can be connected to and from using connect statements.
```

Comment:
can we use the `wire`{.firrtl} syntax?

Reply:
The concept is wire, the syntax is `wire`{.firrtl}. I considered this, and in this context, it's better to separate the concept from the syntax. (Otherwise, "port" becomes `input` and `output` and "registers" becomes `reg` and `regreset`).